python/eos/analysis.py

```diff
@@ -476,8 +480,10 @@ def sample_pmc(self, log_proposal, step_N=1000, steps=10, final_N=5000, rng=np.r
             eos.info('Convergence diagnostics of all previous samples after sampling in step {}: perplexity = {}, ESS = {}'.format(
                 step, self._perplexity(adjusted_weights), self._ess(adjusted_weights)
                 ))
-            pypmc.mix_adapt.pmc.gaussian_pmc(samples, sampler.proposal, adjusted_weights, mincount=0, rb=True, copy=False)
+            pmc = pypmc.mix_adapt.pmc.PMC(samples, sampler.proposal, adjusted_weights, mincount=0, rb=True)
+            pmc.run(iterations=pmc_iterations, prune=0.0, rel_tol=pmc_rel_tol, abs_tol=pmc_abs_tol)
             # Normalize the weights and remove components with a weight smaller than weight_threshold
+            sampler.proposal = pmc.density
             sampler.proposal.normalize()
             sampler.proposal.prune(threshold = weight_threshold)
```

Comment:
This prunes components with weight below ``weight_threshold`` only **after** the PMC updates have been performed. I have no intuition if this is worse than pruning components **during** the update steps. However, a comment in the code would be nice.

Reply:
Hmm no clue, this would need a dedicated study... I would say that the current version is the most conservative but it maybe unnecessarily time consuming...